lib/rubocop/cop/style/if_with_boolean_literal_branches.rb

```diff
@@ -6,6 +6,8 @@ module Style
       # This cop checks for redundant `if` with boolean literal branches.
       # It checks only conditions to return boolean value (`true` or `false`) for safe detection.
       # The conditions to be checked are comparison methods, predicate methods, and double negative.
+      # However, auto-correction is unsafe because there is no guarantee that all predicate methods
+      # will return boolean value.
```

Comment:
I guess that's a good place to mention that's also the reason we have configuration for this cop. :-)

Reply:
Sure! I added the doc for configuration.